recipes-openxt/qemu-dm/qemu-dm-2.6.2/ovmf-q35.patch

```diff
@@ -0,0 +1,284 @@
++       int sector_bits, sector_size;
++       char name[64];
++
++        pflash_drv = drive_get(IF_PFLASH, 0, 0);
++
++       if (!pflash_drv) {
++               error_report("XEN EFI NVRAM - No pflash device found\n");
++               exit(1);
++       }
++
++
++               blk = blk_by_legacy_dinfo(pflash_drv);
++               size = blk_getlength(blk);
++
++       sector_bits = 12;
++       sector_size = 1 << sector_bits;
++
++
++        if (size < 0) {
++            fatal_errmsg = g_strdup_printf("failed to get backing file size");
++        } else if (size == 0) {
++            fatal_errmsg = g_strdup_printf("PC system firmware (pflash) "
++                               "cannot have zero size");
++        } else if ((size % sector_size) != 0) {
++            fatal_errmsg = g_strdup_printf("PC system firmware (pflash) "
++                               "must be a multiple of 0x%x", sector_size);
++        } else if (size != 0x20000) {
++            fatal_errmsg = g_strdup_printf("image must be exactly 0x20000 bytes, yours is 0x%x",(unsigned) size);
++        }
++
++        if (fatal_errmsg != NULL) {
++            Location loc;
++
++            /* push a new, "none" location on the location stack; overwrite its
++             * contents with the location saved in the option; print the error
++             * (includes location); pop the top
++             */
++            loc_push_none(&loc);
++            if (pflash_drv->opts != NULL) {
++                qemu_opts_loc_restore(pflash_drv->opts);
++            }
++            error_report("%s", fatal_errmsg);
++            loc_pop(&loc);
++            g_free(fatal_errmsg);
++            exit(1);
++        }
++
++        /* pflash_cfi01_register() creates a deep copy of the name */
++        snprintf(name, sizeof name, "system.flash0");
++        pflash_cfi01_register(phys_addr, NULL /* qdev */, name,
++                                             size, blk, sector_size,
++                                             size >> sector_bits,
++                                             1      /* width */,
++                                             0x0000 /* id0 */,
++                                             0x0000 /* id1 */,
++                                             0x0000 /* id2 */,
++                                             0x0000 /* id3 */,
++                                             0      /* be */);
++}
++
++void pc_q35_init(MachineState *machine);
++
+ /* PC hardware initialisation */
+-static void pc_q35_init(MachineState *machine)
++void pc_q35_init(MachineState *machine)
+ {
+     PCMachineState *pcms = PC_MACHINE(machine);
+     PCMachineClass *pcmc = PC_MACHINE_GET_CLASS(pcms);
+@@ -143,6 +220,8 @@ static void pc_q35_init(MachineState *ma
+     if (!xen_enabled()) {
+         pc_memory_init(pcms, get_system_memory(),
+                        rom_memory, &ram_memory);
++    } else {
++       xen_map_efi_var_rom(rom_memory);
```

Comment:
Will this logic trigger for non-uefi guest on xen and if so, is it handled gracefully?